src/near_client.rs

```diff
@@ -102,19 +103,31 @@ impl NeardClient {
     ) -> Result<MaintenanceWindowRPCResult> {
         let mut params = HashMap::<String, serde_json::Value>::new();
         params.insert("account_id".into(), account_id.as_str().into());
-        let res = self
-            .client
-            .post(self.url.clone())
-            .json(&Self::rpc_request(
-                "EXPERIMENTAL_maintenance_windows",
-                params,
-            ))
-            .send()
-            .await
-            .context("Failed to get maintenance windows")?;
 
-        let r: MaintenanceWindowJsonRpcStatusResponse = res.json().await?;
-        Ok(r.result)
+        let mut res = None;
+        for _ in 0..3 {
+            if let Ok(r) = self
+                .client
+                .post(self.url.clone())
+                .json(&Self::rpc_request(
+                    "EXPERIMENTAL_maintenance_windows",
+                    params.clone(),
+                ))
+                .send()
+                .await
+            {
+                res = Some(r);
+                break;
+            }
+            sleep_until(Instant::now() + Duration::from_secs(1)).await;
+        }
+
+        if let Some(res) = res {
+            let r: MaintenanceWindowJsonRpcStatusResponse = res.json().await?;
+            Ok(r.result)
+        } else {
+            anyhow::bail!("Failed to get maintenance windows from neard rpc with 3 times retry")
```

Comment:
```suggestion
            anyhow::bail!("Failed to get maintenance windows from neard RPC. Please try again.")
```

But is there no way to know 'why' - what else might the user need to do to debug?

Reply:
I checked the code in nearcore.  This was the code for this error.
https://github.com/near/nearcore/blob/02bd5996d0e581f12768baa9f3f68849a77a8312/chain/jsonrpc/src/lib.rs#L1095
It should be an empty list if there are no windows, other select window logic is not here. The only possible issue here is the node is too busy to get all possible windows in the list.  Anyway, I will keep the error and print it out.